### PR TITLE
add knapsack fraction problem using greedy appraoch.

### PR DESCRIPTION
**Pull request by bagladivyang03**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by bagladivyang03** [comment]
@HarshCasper Could u pls review and merge it :)

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Enter the number of items :5
+Enter Profit and weight for item15 2
+Enter Profit and weight for item23 1
+Enter Profit and weight for item36 3
+Enter Profit and weight for item48 1
+Enter Profit and weight for item510 6
+Enter the capacity of knapsack :
+the maximum profit is : 32
+
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Add a new line at the EOF

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#include<iostream>
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Add a beginner pitch about the Code/Approach

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            ratio[j] = ratio[j+1];
+            ratio[j+1] = temp;
+ 
+            temp = weight[j];
+            weight[j] = weight[j+1];
+            weight[j+1] = temp;
+ 
+            temp = profit[j];
+            profit[j] = profit[j+1];
+            profit[j+1] = temp;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Put apt comments explaining the Code

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        cout<<"Enter Profit and weight for item"<<i+1;
+        cin>>profit[i]>>weight[i];
+    } 
+    cout<<"Enter the capacity of knapsack :"<<endl;
+    cin>>capacity;
+     
+     for(i=0;i<n;i++)
+         ratio[i]=profit[i]/weight[i];
+    
+    sort_arr(ratio,profit,weight,n);
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Inconsistent indentation and formatting

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    
+    
+    int ans = knapsack_fraction(weight,ratio,profit,n,capacity);
+    cout<<"the maximum profit is : "<<ans<<endl;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Use properly formatted console statements

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by bagladivyang03** [comment]
@HarshCasper could u check now

**Comment by bagladivyang03** [comment]
@HarshCasper could u pls check :)

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    cout << "Enter the capacity of knapsack :" << endl;
+    cin >> capacity;
+
+    for (i = 0; i < n; i++) //making the new array and storing ratio (profit/weight)
+        ratio[i] = profit[i] / weight[i];
+
+    sort_arr(ratio, profit, weight, n); //first sort the array acc to Profit/weight
+
+    int ans = knapsack_fraction(weight, ratio, profit, n, capacity); //call the function to calculate max profit.
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 7]
```diff
@@ -0,0 +1,83 @@
+/*
+Given weights and values of n items, we need to put these items in a knapsack of capacity W to get the maximum total value in the knapsack.
+Here We are solving fractional knapsack in which we can take a fraction of item too.
+*/
+
+#include <iostream>
+using namespace std;
+
+void sort_arr(float ratio[], float profit[], float weight[], int n) // A function to sort the arrays according to profit/weight
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                weight[j] = weight[j + 1];
+                weight[j + 1] = temp;
+
+                temp = profit[j];
+                profit[j] = profit[j + 1];
+                profit[j + 1] = temp;
+            }
+}
+
+int knapsack_fraction(float weight[], float ratio[], float profit[], int n, int capacity) //function to calculate max value in knapsack.
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+{
+    int final = 0;
+    int i;
+    for (i = 0; i < n; i++) //iterating over all items
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        if (weight[i] > capacity) //checking if current weight is greater tahn break i.e this weight can't be put into knapsack.
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+            break;
+        else //else add profir to final varibale i.e max profit and update capacity accordingly
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            final = final + profit[i];
+            capacity = capacity - weight[i];
+        }
+    }
+    if (i < n) //for fraction amount of item
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    return final;
+}
+
+int main()
+{
+    float weight[20], profit[20], ratio[20], final, temp, capacity, amount;
+    int n, i, j;
+    cout << "Enter the number of items :";
+    cin >> n;
+    for (i = 0; i < n; i++) //accepting the profit and weight values for given value of n.
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        cout << "Enter Profit and weight for item" << i + 1;
+        cin >> profit[i] >> weight[i];
+    }
+    cout << "Enter the capacity of knapsack :" << endl;
+    cin >> capacity;
+
+    for (i = 0; i < n; i++) //making the new array and storing ratio (profit/weight)
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

File: C-Plus-Plus/other/knapsackfraction.cpp [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        ratio[i] = profit[i] / weight[i];
+
+    sort_arr(ratio, profit, weight, n); //first sort the array acc to Profit/weight
```
Comment (HarshCasper) [inline C-Plus-Plus/other/knapsackfraction.cpp]: Comments should be made above the line of code, not beside it

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny changes needed. Rest looks good to merge.

**Comment by bagladivyang03** [comment]
@HarshCasper Could u pls check now :)

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**